### DISCUSSION TITLE
Incremental static regeneration issue on Vercel

### DISCUSSION THREAD

**Discussion by flaviolivolsi** [vercel/next.js#20102] [Help]
Hello,

I have an issue with the revalidation process and I'd like to ask you whether you can suggest a solution.

I have an application built with Next.js which makes use of the Incremental Static Regeneration in order to regenerate the content without build.  Given this premise, I'll make an example to better explain the issue.

Let's assume I have a page (http://mysite.com/page/1) deployed with "Hello" as title (title being of course fetched from the API called in getStaticProps).
Let's also assume that, after the build, I change the title from "Hello" to "Hello world".

What happens after these step is the following:
- When I visit http://mysite.com/page/1 for the first time after the title change, the old title "Hello" shows up (and the regeneration is triggered)
- When I visit http://mysite.com/page/1 for the second time after the title change, the new title "Hello world" shows up
- When I visit http://mysite.com/page/1?foo=bar after the first two hits at the page without query parameters, the old title "Hello" shows up

Unfortunately this behavior is not ideal for our case. Our traffic is mostly gathered from Facebook Ads, hence the users who visit our pages always carry a GET parameter ("fbclid") which is unique for each visit. As a result, every visitor ends up visiting the old version of every page. Right now we're solving by triggering a build via webhook following every change.

The ideal scenario is: 
- When I visit http://mysite.com/page/1 for the first time after the title change, the old title "Hello" shows up (and the regeneration is triggered)
- When I visit http://mysite.com/page/1 for the second time after the title change, the new title "Hello world" shows up
- When I visit http://mysite.com/page/1?foo=bar after the first two hits at the page without query parameters, the new title "Hello world" shows up

When I run Next.js locally (whether it's in dev or production mode) it works as per my ideal scenario. The issue occurs only on Vercel.

Is there any way I can achieve the behavior I would like?

Thank you!

**Comment by giupas** [comment]
I’ve a similar issue and I’ve find out two possible bugs:
- visiting a page with query parameters trigger the router two times, the first page loaded is without parameters, the second is with parameters. This is not ideal for advertising or anything not 100% compliant with React logic.
- revalidate is not always reliable: the page with parameters does not update because the data json for in app navigation is not updated. In my case some pages won’t update at all for hours. I’ll send a report when I’ll be able to replicate the issue on a sample app.

**Reply by times29** [reply]
I'm facing the problem you mentioned in the second bullet point as well. Have you created an issue yet?

**Reply by ghost** [reply]
Same issue here. 
@giupas here is the source in the docs to your second point https://nextjs.org/docs/basic-features/data-fetching#statically-generates-both-html-and-json

> [...] client-side page transitions will not call getStaticProps as only the exported JSON is used.

**Reply by giupas** [reply]
That's correct, but could lead to inconsistency. The exported json should be no longer valid when its page gets updated, while right now client side uses always the old json. 

> > [...] client-side page transitions will not call getStaticProps as only the exported JSON is used.

**Reply by giupas** [reply]
> I'm facing the problem you mentioned in the second bullet point as well. Have you created an issue yet?

Not yet.

**Reply by hatsumatsu** [reply]
Adding an API to ignore / allow query parameters for ISR would be a great addition to nextjs.

Another negative effect of the current logic: build minutes go through the roof if every unique query parameter triggers the creation of a static page.

Related: https://github.com/vercel/next.js/discussions/24091
